_rules/image-accessible-name-23a2a8.md

```diff
@@ -39,7 +40,7 @@ The rule applies to HTML `img` elements or any HTML element with the [semantic r
 
 ## Expectation
 
-Each target element has an [accessible name][] that is not empty (`""`), or is [marked as decorative][].
+Each target element has an [accessible name][] that is not empty (`""`), or has a [semantic role][] of `none` or `presentation`.
```

Comment:
That is _literally_ just inlining the definition of "marked as decorative".

The definition of "marked as decorative" depends on the semantic role. Changing the definition of "semantic role" does also change "marked as decorative". I think that the name does not really convey the intend of the new definition.

I think that "marked as decorative" should be "explicit role of `none`/`presentation` or (for `img`) empty `alt`". Given that it was only really used here, I though it would be better to inline the definition and remove one glossary item.

I am indeed not entirely sure whether that rule should check "accessible name or role of `presentation`" or "accessible name, or explicit role of `presentation`, or empty `alt`"

The difference between the two would be for `<img tabindex=1 role="none" />` which would fail the "accessible name or role of `presentation`" version (with resolution as implemented by Chrome and Firefox) but pass the "accessible name, or explicit role of `presentation`, or empty `alt`" (on all browsers).

I think this image creates an accessibility issue (because it is tabbable but has no name) and thus chose the version that fails it…

I am happy to revisit if you consider that this should pass the rule instead.

Reply:
I agree with the change, now that you've changed what semantic role does with elements not in the accessibility tree. Still, I think this should be in a separate PR. It's a completely unrelated problem.